### PR TITLE
PV-4867 fixes for GP master list

### PR DESCRIPTION
**Pull request by p-maks**

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/service/impl/GpServiceImpl.java [hunk 1]
```diff
@@ -823,52 +824,28 @@ private void updateNorthernIreland() throws IOException, ZipException {
...
-                        }
+                    "/" + GpCountries.NI.toString() + "/" + GpCountries.NI.toString() + ".csv"));
 
-                        addToSaveMap(practiceCode, practiceName, address1, address2, address3, null,
-                                postcode, null, telephone, GpCountries.NI);
-                    }
+            FileUtils.copyURLToFile(new URL(url), extractedDataFile);
+
+            // read CSV file line by line, extracting data to populate GpMaster objects
+            CSVParser parser = new CSVParser(new FileReader(extractedDataFile), CSVFormat.DEFAULT);
```
Comment (alexpask) [inline root/api/src/main/java/org/patientview/api/service/impl/GpServiceImpl.java]: Could use a try-with-resources here

File: root/api/src/main/resources/conf/local-api.properties [hunk 2]
```diff
@@ -90,16 +90,15 @@ nhschoices.api.key=bbccdd
 nhschoices.conditions.api.key=
 nhschoices.organisation.api.key=
 
-# GP master table and import
-gp.master.temp.directory=/Users/james/patientview/gpmaster
-gp.master.url.england=http://systems.hscic.gov.uk/data/ods/datadownloads/data-files/epraccur.zip
+# GP master table and import, changes quarterly
+gp.master.temp.directory=/Users/pmaksymchuk/patientview/gpmaster
```
Comment (alexpask) [inline root/api/src/main/resources/conf/local-api.properties]: Is it possible to change this to a generic directory like /tmp or something

File: root/api/src/main/resources/conf/local-api.properties [hunk 3]
```diff
@@ -90,16 +90,15 @@ nhschoices.api.key=bbccdd
...
 gp.master.filename.england=epraccur.csv
-gp.master.url.scotland=http://www.isdscotland.org/Health-Topics/General-Practice/Workforce-and-Practice-Populations/Practices-and-Their-Populations/_docs/Practice_ContactDetails_Jan2016_final.xls
-# changes monthly, see 'Northern Ireland GP \ Practice lists for professional use' at http://www.hscbusiness.hscni.net/services/1816.htm
-gp.master.url.northernireland=http://www.hscbusiness.hscni.net/pdf/Northern_Ireland_Practice_List_010216.xls
+gp.master.url.scotland=https://www.opendata.nhs.scot/dataset/f23655c3-6e23-4103-a511-a80d998adb90/resource/34f02dbe-2827-47ae-821f-d529e26075cd/download/gppractices20181001.csv
+gp.master.url.northernireland=https://www.opendatani.gov.uk/dataset/3d1a6615-5fc9-4f0e-ab2a-d2b0d71fb9ed/resource/59e8bffd-9faa-45f2-8d7c-9a2bdff8a147/download/gp-practice-reference-file---october-2018.csv
 
 # GP letters export
-gp.letter.output.directory=/Users/james/patientview/gpletter
+gp.letter.output.directory=/Users/pmaksymchuk/patientview/gpletter
```
Comment (alexpask) [inline root/api/src/main/resources/conf/local-api.properties]: Same as the above comment here

File: root/api/src/main/java/org/patientview/api/service/impl/GpServiceImpl.java (hunk 1 above)
Comment (p-maks) [reply root/api/src/main/java/org/patientview/api/service/impl/GpServiceImpl.java]: I need it to throw exception if anything goes wrong

File: root/api/src/main/resources/conf/local-api.properties (hunk 2 above)
Comment (p-maks) [reply root/api/src/main/resources/conf/local-api.properties]: updated

File: root/api/src/main/resources/conf/local-api.properties (hunk 3 above)
Comment (p-maks) [reply root/api/src/main/resources/conf/local-api.properties]: updated

**Pull request merged**